### DISCUSSION TITLE
Create GCC toolchain (tar ball in xtensa-overlay repo) to a new platform.

### DISCUSSION THREAD

**Discussion by saisurya-ch** [thesofproject/sof#9324] [Q&A] [answered]
Hi all,

To support the SOF GCC build, we have the repositories [xtensa-ovelay](https://github.com/thesofproject/xtensa-overlay) & [crosstool-ng](https://github.com/thesofproject/crosstool-ng) . I would like to create a tar ball for a new AMD platform. I would like to know the flow of creating the platform specific tar ball & the lib files those need to be added to the tar.gz (which is in xtensa-overlay repo).

Facing this below issue while the GCC build

[ 20%] Building ASM object src/arch/xtensa/xtos/CMakeFiles/xtos.dir/core-shutoff.S.o
[ 20%] Performing configure step for 'smex_ep'
[ 21%] Building ASM object src/arch/xtensa/xtos/CMakeFiles/xtos.dir/double-vector.S.o
[ 21%] Building ASM object src/arch/xtensa/xtos/CMakeFiles/xtos.dir/debug-vector.S.o
[ 21%] Linking C static library libhal.a
[ 22%] Performing configure step for 'rimage_ep'
**sof/src/arch/xtensa/xtos/core-restore.S: Assembler messages:
sof/src/arch/xtensa/xtos/core-restore.S:235: Error: unknown opcode or format name 'wur.xnne_rerr'
gmake[3]: *** [src/arch/xtensa/xtos/CMakeFiles/xtos.dir/build.make:76: src/arch/xtensa/xtos/CMakeFiles/xtos.dir/core-restore.S.o] Error 1
gmake[3]: *** Waiting for unfinished jobs....**
[ 22%] Built target reset
[ 22%] Built target ext_manifest
[ 22%] Built target hal
**sof/src/arch/xtensa/xtos/core-save.S: Assembler messages:
sof/src/arch/xtensa/xtos/core-save.S:554: Error: unknown opcode or format name 'rur.xnne_rerr'
gmake[3]: *** [src/arch/xtensa/xtos/CMakeFiles/xtos.dir/build.make:90: src/arch/xtensa/xtos/CMakeFiles/xtos.dir/core-save.S.o] Error 1
[ 22%] Linking ASM static library libxlevel4.a
[ 22%] Linking ASM static library libxlevel5.a
gmake[2]: *** [CMakeFiles/Makefile2:2090: src/arch/xtensa/xtos/CMakeFiles/xtos.dir/all] Error 2
gmake[2]: *** Waiting for unfinished jobs....**
[ 22%] Linking ASM static library libxlevel3.a
[ 22%] Built target xlevel4
[ 22%] Built target xlevel5
[ 22%] Built target xlevel3
-- The C compiler identification is GNU 11.4.0
-- The C compiler identification is GNU 11.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info

**Comment by lgirdwood** [comment, marked as answer]
```
sof/src/arch/xtensa/xtos/core-save.S: Assembler messages:
sof/src/arch/xtensa/xtos/core-save.S:554: Error: unknown opcode or format name 'rur.xnne_rerr'
gmake[3]: *** [src/arch/xtensa/xtos/CMakeFiles/xtos.dir/build.make:90: src/arch/xtensa/xtos/CMakeFiles/xtos.dir/core-
```
This is caused by the supplied binutils patches not being aligned with your ISA config, 2 fixes
1) Raise a ticket with Cadence and get the correct binutils patches.

OR

2) Work around this problem with an ifdef for GCC. GCC is not building runtime binaries (no hifi SIMD support), but is building for CI sanity and static analysis reasons.
